[resolver] show possible candidates when resolving a version contrained requirement failed

Assume I get the following error from the resolver:

```
[ERROR] Resolution failed. Capabilities satisfying the following requirements could not be found:
    [<<INITIAL>>]
      ⇒ osgi.identity: (&(osgi.identity=my.test.fragment)(version>=1.0.0))
          ⇒ [my.test.fragment version=1.0.0 type=osgi.fragment]
              ⇒ osgi.wiring.package: (osgi.wiring.package=org.osgi.test.junit5.service)
                  ⇒ [org.osgi.test.junit5 version=1.2.1]
                      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.osgi.framework)(version>=1.8.0)(!(version>=2.0.0)))
    [org.apache.felix.scr version=2.2.4]
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.osgi.dto)(version>=1.1.0)(!(version>=2.0.0)))
```

the currently it is hard to guess what might be missing without carefully inspection of the used repositories, e.g.

- The package *might* exists, but the version do not match
- The package is not available at all.

It would be good if the resolver when it encounters some "well-known errors" (e.g. a missing package or identity ) it does the following:

1.  Query for the package/identity **without** a version, and print a lit of versions that could be found instead
2. If not package/identity is found, print that no available bundle export the package or no identity with that name is found

Even though this only works for same special cases, these are the most common ones and thus has a very great value to the user as errors in the setup are much easier to spot.

--- a/biz.aQute.resolve/src/biz/aQute/resolve/Bndrun.java
+++ b/biz.aQute.resolve/src/biz/aQute/resolve/Bndrun.java
@@ -107,6 +107,7 @@ public class Bndrun extends Run {
 	public <T> T resolve(boolean failOnChanges, boolean writeOnChanges,
 		Converter<T, Collection<? extends HeaderClause>> runbundlesFormatter) throws Exception {
 
+		checkValidate();
 		RunResolution resolution = RunResolution.resolve(this, this, null);
 
 		if (!resolution.isOK()) {
@@ -117,6 +118,7 @@ public class Bndrun extends Run {
 	}
 
 	public RunResolution resolve(ResolutionCallback... callbacks) throws Exception {
+		checkValidate();
 		RunResolution resolution = RunResolution.resolve(this, this, Arrays.asList(callbacks))
 			.reportException();
 		if (!resolution.isOK()) {
@@ -304,6 +306,16 @@ public class Bndrun extends Run {
 			}
 		}
 		return true;
+	}
+
+	private void checkValidate() {
+		if (getProperty(Constants.RUNFRAMEWORK) != null && getProperty(Constants.RUNFW) == null) {
+			warning("""
+				The `-runframework` instruction is set and not the `-runfw`. This is a very unlikely
+				combination. To specify the framework for the resolution, use `-runfw`, the use of `
+				-runframework` is quite esoteric and related to testing.
+				""");
+		}
 	}
 
 }

